crates/uplc/src/parser.rs

```diff
@@ -251,7 +251,14 @@ peg::parser! {
             PlutusData::Map(pallas_codec::utils::KeyValuePairs::Def(kvps))
           }
           / _* "List" _+ ls:plutus_list() { PlutusData::Array(ls) }
-          / _* "I" _+ n:number() {? Ok(PlutusData::BigInt(pallas_primitives::babbage::BigInt::Int(i64::try_from(n).or(Err("int"))?.into()))) }
+          / _* "I" _+ n:big_number() { match i64::try_from(n.clone()) {
```

Comment:
Should use to_i64 instead of try from, I believe that is clearer. 
And use sign to match on positive or negative instead of the < 0 check.